src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -411,6 +422,120 @@ namespace winrt::Microsoft::Terminal::Control::implementation
         }
     }
 
+    // Method Description:
+    // - Updates the settings
+    // - This should only be called from the UI thread
+    // Arguments:
+    // - newSettings: the new settings to set
+    void TermControl::_UpdateSettingsFromUIThread(IControlSettings newSettings)
+    {
+        if (_closing)
+        {
+            return;
+        }
+
+        // Update our control settings
+        _ApplyUISettings(_settings);
+
+        // Update the terminal core with its new Core settings
+        _terminal->UpdateSettings(_settings);
+
+        auto lock = _terminal->LockForWriting();
+
+        // Update DxEngine settings under the lock
+        _renderEngine->SetRetroTerminalEffect(_settings.RetroTerminalEffect());
+        _renderEngine->SetForceFullRepaintRendering(_settings.ForceFullRepaintRendering());
+        _renderEngine->SetSoftwareRendering(_settings.SoftwareRendering());
+
+        switch (_settings.AntialiasingMode())
+        {
+        case TextAntialiasingMode::Cleartype:
+            _renderEngine->SetAntialiasingMode(D2D1_TEXT_ANTIALIAS_MODE_CLEARTYPE);
+            break;
+        case TextAntialiasingMode::Aliased:
+            _renderEngine->SetAntialiasingMode(D2D1_TEXT_ANTIALIAS_MODE_ALIASED);
+            break;
+        case TextAntialiasingMode::Grayscale:
+        default:
+            _renderEngine->SetAntialiasingMode(D2D1_TEXT_ANTIALIAS_MODE_GRAYSCALE);
+            break;
+        }

```

Comment:
Wait does this whole chunk of work still need to be done? Isn't it going to be done in the subsequent call to `_UpdateAppearanceFromUIThread`?

(presumably, taking the lock does need to be done for `_RefreshSizeUnderLock`)

Reply:
Deleted `SetRetroTerminalEffect` from here, everything else does need to be here because its not done in `_UpdateAppearanceFromUIThread`